service/gcs/main.go

```diff
@@ -15,21 +17,40 @@ import (
 )
 
 func main() {
-	// parse command line parameters and init logger
-	if err := utils.ProcessCommandlineOptions(); err != nil {
-		logrus.Fatalf("%+v", err)
+	logLevel := flag.String("loglevel", "warning", "Logging Level: debug, info, warning, error.")
+	logFile := flag.String("logFile", "", "Logging Target: An optional file name/path. Omit for console output.")
+
+	flag.Usage = func() {
+		fmt.Fprintf(os.Stderr, "\nUsage of %s:\n", os.Args[0])
+		flag.PrintDefaults()
+		fmt.Fprintf(os.Stderr, "Examples:\n")
+		fmt.Fprintf(os.Stderr, "    %s -loglevel=debug -logfile=gcslog.txt (default)\n", os.Args[0])
+		fmt.Fprintf(os.Stderr, "    %s -loglevel=info -logfile=stdout\n", os.Args[0])
+	}
+
+	flag.Parse()
+
+	// Use a file instead of stdout
+	if *logFile != "stdout" {
+		logFileHandle, err := os.OpenFile(*logFile, os.O_CREATE|os.O_WRONLY, 0600)
+		if err != nil {
+			logrus.Fatalf("%+v", errors.Errorf("failed to create log file %s", *logFile))
```

Comment:
You can probably just use logrus.Fatal and pass in only the error to it here. The %+v is for when an error returned is a github.com/pkg/errors error, where %+v means "print out the error message and its stack trace". Since os.OpenFile doesn't return an error with a stack trace, the %+v isn't necessary.